Python/Product/Analysis/LanguageServer/Server.Hover.cs

```diff
@@ -76,29 +57,71 @@ public sealed partial class Server {
             }
 
             TraceMessage($"Getting hover for {expr.ToCodeString(tree, CodeFormattingOptions.Traditional)}");
-            var values = analysis.GetValues(expr, @params.position, scope).ToList();
 
-            string originalExpr;
-            if (expr is ConstantExpression || expr is ErrorExpression) {
-                originalExpr = null;
+            // First try values from expression. This works for 
+            // import statements most of the time as well.
+            var index = tree.LocationToIndex(@params.position);
+            var w = new ImportedModuleNameWalker(entry, index, tree);
+            tree.Walk(w);
+
+            IEnumerable<AnalysisValue> values = null;
+            // Perhaps it is an assigned member like 'p' in 'from os import path as p'
+            if (!string.IsNullOrEmpty(w.ImportedType?.Name)) {
+                values = analysis.GetValues(w.ImportedType.Name, @params.position).ToList();
             } else {
-                originalExpr = @params._expr?.Trim();
-                if (string.IsNullOrEmpty(originalExpr)) {
-                    originalExpr = expr.ToCodeString(tree, CodeFormattingOptions.Traditional);
+                // Perhaps it is a member like 'path' in 'from os import path'
```

Comment:
In this and the previous case, they are properly defined variables and our normal handling will be just fine. We only need special handling for the module name (which is not a variable in the current file) and the imported name when it gets reassigned (the `path` in `from os import path as p`) - in the latter case, we should be able to easily set `expr` to the "AsName" from the import statement and it'll show the right content (even better, it'll show the name the user _ought_ to be using to access it).

Reply:
`analysis.GetValues` does not return anything for `path`. But yeah, we can try `AsNames`.

Reply:
Right, because `path` isn't a real variable in this file. So we need to handle that one specially, but not `p`.